src/server/command_registry.cc

```diff
@@ -48,6 +50,18 @@ bool CommandId::IsTransactional() const {
   return false;
 }
 
+void CommandId::Invoke(CmdArgList args, ConnectionContext* cntx) const {
+  uint64_t before = absl::GetCurrentTimeNanos();
+  const char* const cid =
+      cntx->cid->name().data();  // this is OK as we assume the name is a literal
```

Comment:
I noticed an issue with EXEC which doesn't restore its own CID after returning, causing double counting. This is a patch, I'll look more before fixing to make sure I'm not missing anything.